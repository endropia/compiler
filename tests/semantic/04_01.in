var
	a: integer;
	b: double;
	c: boolean;
begin
	writeln(+a);
	writeln(+b);

	writeln(-a);
	writeln(-b);

	writeln(not a);
	writeln(not c);
end.