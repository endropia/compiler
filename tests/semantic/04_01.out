program : Unnamed program
   var: 
      a
      type: integer
   var: 
      b
      type: double
   var: 
      c
      type: boolean
   stmts:
      call
         writeln
            +
            a
      call
         writeln
            +
            b
      call
         writeln
            -
            a
      call
         writeln
            -
            b
      call
         writeln
            not
            a
      call
         writeln
            not
            c

scope     name                          class               
------------------------------------------------------------
0         integer                       primitive type      
0         double                        primitive type      
0         boolean                       primitive type      
0         char                          primitive type      
0         string                        primitive type      
1         a                             variable            
1         b                             variable            
1         c                             variable            
